examples/games/mafia_werewolf/play_mafia.qnt

```diff
@@ -0,0 +1,185 @@
+// -*- mode: Bluespec; -*-
+
+module mafia {
+    const PLAYERS : Set[str]
+    // Types for roles and life states of players
+    type Role = Mafia | Citizen
+    type LifeState = Alive | Dead
+    type Phase = Day | Night
+    type PlayerFeatures = {
+        role: Role,
+        status: LifeState,
+        voted: bool  // Indicates whether the player has voted
+    }
+    var player_to_features: str -> PlayerFeatures
+    var votes_by_player: str -> int
+    var game_phase: Phase
+    type Status = Pending | Done(Role)
+    var game_status: Status
+    /// Check if all alive players have voted
+    val all_voted = PLAYERS.filter(p => player_to_features.get(p).status == Alive)
+                      .forall(p => player_to_features.get(p).voted == true)
+    /// Check if there are any Mafia players left
+    val has_mafia = PLAYERS.exists(p => player_to_features.get(p).role == Mafia)
+    /// Check if there are any Citizen players left
+    val has_citizen = PLAYERS.exists(p => player_to_features.get(p).role == Citizen)
+    /// Find players with the most votes if all players have voted
+    val get_most_voted_players = {
+        if (all_voted) {
+            val max_votes = PLAYERS.fold(-1, (acc, p) => {
+                val votes = votes_by_player.get(p)
+                if (votes > acc) votes else acc
+            })
+            PLAYERS.filter(p => votes_by_player.get(p) == max_votes)
+        } else Set()  // Return an empty set if not all players have voted
+    }
+    /// Check if all Mafia players are dead
+    pure def all_mafias_dead(player: str -> PlayerFeatures): bool = {
+        PLAYERS.filter(p => player.get(p).role == Mafia).forall(p => player.get(p).status == Dead)
+    }
```

Comment:
We could rewrite this (and the next one) into something more compact by using `.values()` and `implies`
```suggestion
    pure def all_mafias_dead(players: str -> PlayerFeatures): bool = {
        players.values().forall(p => p.role == Mafia implies p.status == Dead)
    }
```